src/kbmod/work_unit.py

```diff
@@ -183,18 +227,22 @@ def from_fits(cls, filename):
 
             # Read the size and order information from the primary header.
             num_images = hdul[0].header["NUMIMG"]
-            if len(hdul) != 4 * num_images + 3:
+            n_constituents = hdul[0].header["NCON"]
+            if len(hdul) != (4 * num_images) + (2 * n_constituents) + 3:
                 raise ValueError(
                     f"WorkUnit wrong number of extensions. Expected "
-                    f"{4 * num_images + 3}. Found {len(hdul)}."
+                    f"{6 * num_images + 3}. Found {len(hdul)}."
```

Comment:
Should this be "4 * num_images + 2 * n_constituents + 3" to match the if statement.